client/lib/domains/utils.js

```diff
@@ -46,7 +46,7 @@ function getDomainNameFromReceiptOrCart( receipt, cart ) {
 		domainRegistration = cartItems.getDomainRegistrations( cart )[ 0 ];
 	}
 
-	if ( domainRegistration ) {
+	if ( domainRegistration && ( ! receipt || isEmpty( receipt.failed_purchases ) ) ) {
```

Comment:
Truth is, we use it to check if a `domainName` is there in `checkout.jsx`. I decided to add the check here, instead of adding two additional checks in `checkout.jsx`, as we want to handle failed purchases differently.